src/MongoDbEventStoreAdapter.php

```diff
@@ -254,17 +268,15 @@ public function loadEventsByMetadataFrom(StreamName $streamName, array $metadata
 
             $createdAt = new \DateTime();
             $createdAt->setTimestamp($eventData['created_at']->sec);
-
-            $events[] = $eventClass::fromArray(
-                [
-                    'uuid' => $eventData['_id'],
-                    'name' => $eventData['event_name'],
-                    'version' => (int) $eventData['version'],
-                    'created_at' => $createdAt->format(\DateTime::ISO8601),
-                    'payload' => $eventData['payload'],
-                    'metadata' => $metadata
-                ]
-            );
+            $createdAt->setTimezone(new \DateTimeZone('UTC'));
+
+            $events[] = $this->messageFactory->createMessageFromArray($eventData['event_name'], [
+                'uuid' => $eventData['_id'],
+                'version' => (int) $eventData['version'],
+                'created_at' => $createdAt->format(\DateTime::ISO8601),
```

Comment:
\MongoDate stores already with microseconds. As events can be time-critical, I would tend to inc. microseconds by default.
Whatever the decision will be, let's create another issue for it and merge this one first.